README.md

```diff
@@ -44,7 +44,11 @@ address. The path is discarded.
 
 Options:
 
-- `timeout`, the timeout after which a request will time out, in
+- `socketTimeout`, the timeout after which a socket will time out, in
+  milliseconds.
+  Default: `30e3` milliseconds (30s).
+
+- `requestTimeout`, the timeout after which a request will time out, in
   milliseconds.
```

Comment:
I don't think we should have this enabled by default. Also I think we need to be a bit more explicit about what exactly a request timeout is.

Reply:
I agree that the description should be improved, while regarding the timeout I disagree. From my experience with Fastify, it is always better to have a sane default (30s) rather than not having one at all.
We had a similar issue with the Plugin timeout where we were waiting indefinitely.

Reply:
We do have a timeout on the socket. Shouldn't that be enough?

Reply:
I'm a little worried with the request timeout thing in the cases where the user spams with requests, due to pipelining they might never finish within the request timeout, and then the user keeps retrying them.

Reply:
@mcollina: opinion, `requestTimeout` enabled by default or not?

Reply:
I would prefer to have it because in the end every use will configure it at some point.
But if you think that it's overkill, I'm also fine not having it.
Still, there should be a way to configure it both globally and for a single request.

Reply:
> Still, there should be a way to configure it both globally and for a single request.

Agreed. I'm mostly concerned with the default.

Reply:
> opinion, requestTimeout enabled by default or not?

As long as we support `requestTimeout: 0` to disable it, I do not see a difference. Probably it's safer to keep a default so a long running request on the server will not have a pipeline starve.

Reply:
Agreed! I’ll work on this :)

Reply:
Done 🎉